extensions/smallrye-health/deployment/src/test/java/io/quarkus/smallrye/health/test/ExpectedBeansUnitTest.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.smallrye.health.test;
+
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.Liveness;
+import org.eclipse.microprofile.health.Readiness;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.wildfly.common.Assert;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.health.HealthGroup;
+import io.smallrye.health.SmallRyeHealthReporter;
+
+public class ExpectedBeansUnitTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(FailingHealthCheck.class)
+                    .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
+    @Inject
+    @Any
+    Instance<HealthCheck> checks;
+
+    @Inject
+    Instance<SmallRyeHealthReporter> reporters;
+
+    private boolean isUnique(Instance<?> instances) {
+        return !(instances.isAmbiguous() && instances.isUnsatisfied());
```

Comment:
I suppose that by "is unique" you mean "is exactly one" right? In that case this should probably be
`!instances.isAmbiguous() && !instances.isUnsatisfied()`
Otherwise you would get `true` here even if the `Instance` is ambiguous or unsatisfied (the only way to get `false` returned would be that the `Instance` is ambiguous and unsatisfied at the same time, which is a nonsense), therefore the test would pass even if there were two eligible checks